api/src/main/java/org/openmrs/validator/CohortValidator.java

```diff
@@ -0,0 +1,48 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ * <p>
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.validator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.openmrs.Cohort;
+import org.openmrs.CohortMembership;
+import org.openmrs.Patient;
+import org.openmrs.annotation.Handler;
+import org.springframework.validation.Errors;
+import org.springframework.validation.Validator;
+
+/**
+ * Validates {@link Cohort} objects.
+ */
+@Handler(supports = {CohortMembership.class}, order=50)
+public class CohortValidator implements Validator {
+
+    // Log for this class
+    protected static final Log log = LogFactory.getLog(ConceptMapTypeValidator.class);
+
+    public boolean supports(Class c) {
+        return CohortMembership.class.isAssignableFrom(c);
+    }
+
+    public void validate(Object obj, Errors errors) {
+        if (obj == null || !(obj instanceof CohortMembership)) {
+            throw new IllegalArgumentException("The parameter obj should not be null and must be of type"
+                    + CohortMembership.class);
+        }
+
+        CohortMembership cohortMembership = (CohortMembership) obj;
+        Patient p = cohortMembership.getPatient();
+        if (p.getVoided() != cohortMembership.getVoided()) {
```

Comment:
Imagine that I have a cohort X, with an existing cohort membership linked to patient A who is an active one (not voided), I choose to void the membership and save the cohort, this is a valid cohort and it should pass validation but according to your if clause above that checks `(p.getVoided() != cohortMembership.getVoided()) ` this cohort will get rejected because it will pass this condition since  p.getVoided() will return false while cohortMembership.getVoided() will return true, makes sense?